src/Middleware/ResourceServerMiddleware.php

```diff
@@ -55,7 +55,7 @@ public function __invoke(
         ServerRequestInterface $request,
         ResponseInterface $response,
         callable $out = null
```

Comment:
Here, you can rename $out to $next (it's clearer) and assume it's not null (normally most middleware framework including Expressive will make sure it's never null). It therefore allows to simplify this code: https://github.com/basz/zfr-oauth2-server/blob/refactoring-middleware/src/Middleware/ResourceServerMiddleware.php#L69

```php
$response = $response->withAttribute('oauth_token', $token);
return $next($request, $response);
```

You can actually do that in all middleware and force $next to not be null.